server/src/main/java/org/apache/iotdb/db/protocol/rest/impl/GrafanaApiServiceImpl.java

```diff
@@ -141,6 +144,15 @@ public Response expression(ExpressionRequest expressionRequest, SecurityContext
 
       PhysicalPlan physicalPlan = basicServiceProvider.getPlanner().parseSQLToPhysicalPlan(sql);
 
+      if (physicalPlan instanceof QueryPlan) {
```

Comment:
We'd better not add the limit here. This may cause unexpected behavior when using the grafana plugin.